### PR TITLE
Updated module: added advertise key and fixed inherit peer-policy CLI

### PR DESCRIPTION
**Pull request by PadminiSivaraj**
##### SUMMARY
- Added advertise key
-  Fixed inherit peer-policy CLI

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ios_bgp_address_family

### CODE REVIEW

**Comment by KB-perByte** [comment]
Hey @PadminiSivaraj I was testing out the contribution it failed with the following error,

```
TASK [ios_bgp_address_family : Merge provided configuration with device configuration] ***
task path: /home/thanos/Work/testingStat/collections/ansible_collections/cisco/ios/tests/output/.tmp/integration/ios_bgp_address_family-ik5rfl4h-ÅÑŚÌβŁÈ/tests/integration/targets/ios_bgp_address_family/tests/cli/merged.yaml:9
Loading collection ansible.utils from /home/thanos/Work/testingStat/collections/ansible_collections/ansible/utils
<host> attempting to start connection
<host> using connection plugin ansible.netcommon.network_cli
Found ansible-connection at path /home/thanos/MyVnvs/dev310A213/bin/ansible-connection
<host> found existing local domain socket, using it!
<host> updating play_context for connection
<host>
<host> local domain socket path is /home/thanos/.ansible/pc/5c90c8ce81
<host> Using network group action cisco.ios.ios for cisco.ios.ios_bgp_address_family
<host> ANSIBLE_NETWORK_IMPORT_MODULES: enabled
<host> ANSIBLE_NETWORK_IMPORT_MODULES: found cisco.ios.ios_bgp_address_family  at /home/thanos/Work/testingStat/collections/ansible_collections/cisco/ios/plugins/modules/ios_bgp_address_family.py
<host> ANSIBLE_NETWORK_IMPORT_MODULES: running cisco.ios.ios_bgp_address_family
<host> ANSIBLE_NETWORK_IMPORT_MODULES: complete
fatal: [host]: FAILED! => {
    "changed": false,
    "module_stderr": "advertise l2vpn evpn\r\nadvertise l2vpn evpn\r\n  ^\r\n% Invalid input detected at '^' marker.\r\n\r\nR1(config-router)#",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}
```
Could you please check and correct the integration tests?
Regards.

**Comment by PadminiSivaraj** [comment]
Hello Sagar,
I get the following error 
"Request failed due to following response errors:
 - Could not resolve to a ProjectV2 with the number 8[6]."
 Can you shed some light on how to resolve this?
 Thanks

**Comment by PadminiSivaraj** [comment]
> Hey @PadminiSivaraj I was testing out the contribution it failed with the following error,
> 
> ```
> TASK [ios_bgp_address_family : Merge provided configuration with device configuration] ***
> task path: /home/thanos/Work/testingStat/collections/ansible_collections/cisco/ios/tests/output/.tmp/integration/ios_bgp_address_family-ik5rfl4h-ÅÑŚÌβŁÈ/tests/integration/targets/ios_bgp_address_family/tests/cli/merged.yaml:9
> Loading collection ansible.utils from /home/thanos/Work/testingStat/collections/ansible_collections/ansible/utils
> <host> attempting to start connection
> <host> using connection plugin ansible.netcommon.network_cli
> Found ansible-connection at path /home/thanos/MyVnvs/dev310A213/bin/ansible-connection
> <host> found existing local domain socket, using it!
> <host> updating play_context for connection
> <host>
> <host> local domain socket path is /home/thanos/.ansible/pc/5c90c8ce81
> <host> Using network group action cisco.ios.ios for cisco.ios.ios_bgp_address_family
> <host> ANSIBLE_NETWORK_IMPORT_MODULES: enabled
> <host> ANSIBLE_NETWORK_IMPORT_MODULES: found cisco.ios.ios_bgp_address_family  at /home/thanos/Work/testingStat/collections/ansible_collections/cisco/ios/plugins/modules/ios_bgp_address_family.py
> <host> ANSIBLE_NETWORK_IMPORT_MODULES: running cisco.ios.ios_bgp_address_family
> <host> ANSIBLE_NETWORK_IMPORT_MODULES: complete
> fatal: [host]: FAILED! => {
>     "changed": false,
>     "module_stderr": "advertise l2vpn evpn\r\nadvertise l2vpn evpn\r\n  ^\r\n% Invalid input detected at '^' marker.\r\n\r\nR1(config-router)#",
>     "module_stdout": "",
>     "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
> }
> ```
> 
> Could you please check and correct the integration tests? Regards.

Sagar, I dont face this issue. Can you provide more info.

**Comment by KB-perByte** [comment]
@PadminiSivaraj Thanks for helping me with the test log,
The ios_bgp_global code expects a remote-as configuration before any other neighbor configuration is added, and it seems from the l2 appliance that the dependency is removed.
https://github.com/ansible-collections/cisco.ios/pull/961
Then integration tests should pass once this PR is merged.

Regards

File: tests/integration/targets/ios_bgp_address_family/tests/cli/_remove_config.yaml [hunk 1]
```diff
@@ -18,6 +18,11 @@
 #   ansible.netcommon.cli_config:
 #     config: "{{ lines }}"
 
+- name: Some test data remove later
```
Comment (KB-perByte) [inline tests/integration/targets/ios_bgp_address_family/tests/cli/_remove_config.yaml:21]: This change can be removed, thanks for helping me narrow it down.

Comment (PadminiSivaraj) [reply tests/integration/targets/ios_bgp_address_family/tests/cli/_remove_config.yaml]: Sure. Thanks, Sagar.

**Comment by KB-perByte** [comment]
Hey @PadminiSivaraj the bgp_global issue being fixed we ran into another issue regarding the tests.
logs - 

```
TASK [ios_bgp_address_family : Populate BGP address family configuration] ******
task path: /home/cisco/test_collections/collections/ansible_collections/cisco/ios/tests/output/.tmp/integration/ios_bgp_address_family-_mbkkrby-ÅÑŚÌβŁÈ/tests/integration/targets/ios_bgp_address_family/tests/cli/_populate_config.yaml:4
Loading collection ansible.utils from /home/cisco/test_collections/collections/ansible_collections/ansible/utils
<198.18.1.15> attempting to start connection
<198.18.1.15> using connection plugin ansible.netcommon.network_cli
Found ansible-connection at path /home/cisco/ansible/bin/ansible-connection
<198.18.1.15> found existing local domain socket, using it!
<198.18.1.15> updating play_context for connection
<198.18.1.15> Loading collection ansible.builtin from 
<198.18.1.15> local domain socket path is /home/cisco/.ansible/pc/c7b6748ff8
<198.18.1.15> Using network group action cisco.ios.ios for cisco.ios.ios_bgp_address_family
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: enabled
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: found cisco.ios.ios_bgp_address_family  at /home/cisco/test_collections/collections/ansible_collections/cisco/ios/plugins/modules/ios_bgp_address_family.py
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: running cisco.ios.ios_bgp_address_family
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: complete
fatal: [198.18.1.15]: FAILED! => {
    "changed": false,
    "module_stderr": "address-family ipv4 multicast\r\naddress-family ipv4 multicast\r\n ^\r\n% Invalid input detected at '^' marker.\r\n\r\nc9k-4(config)#",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}
```
could you please check?
Regards

**Comment by KB-perByte** [comment]
@PadminiSivaraj there are still some failures with the integration tests, adding the logs here
```
TASK [ios_bgp_address_family : Populate BGP address family configuration] ******
task path: /home/cisco/test_collections/collections/ansible_collections/cisco/ios/tests/output/.tmp/integration/ios_bgp_address_family-npx0g4_x-ÅÑŚÌβŁÈ/tests/integration/targets/ios_bgp_address_family/tests/cli/_populate_config.yaml:4
Loading collection ansible.utils from /home/cisco/test_collections/collections/ansible_collections/ansible/utils
<198.18.1.15> attempting to start connection
<198.18.1.15> using connection plugin ansible.netcommon.network_cli
Found ansible-connection at path /home/cisco/ansible/bin/ansible-connection
<198.18.1.15> found existing local domain socket, using it!
<198.18.1.15> updating play_context for connection
<198.18.1.15> Loading collection ansible.builtin from 
<198.18.1.15> local domain socket path is /home/cisco/.ansible/pc/91eb0c9053
<198.18.1.15> Using network group action cisco.ios.ios for cisco.ios.ios_bgp_address_family
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: enabled
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: found cisco.ios.ios_bgp_address_family  at /home/cisco/test_collections/collections/ansible_collections/cisco/ios/plugins/modules/ios_bgp_address_family.py
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: running cisco.ios.ios_bgp_address_family
<198.18.1.15> ANSIBLE_NETWORK_IMPORT_MODULES: complete
fatal: [198.18.1.15]: FAILED! => {
    "changed": false,
    "module_stderr": "address-family ipv4 multicast\r\naddress-family ipv4 multicast\r\n ^\r\n% Invalid input detected at '^' marker.\r\n\r\nc9k-4(config)#",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}
```
Regards

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**